recipes/createCredentials-to-createSecureContext/src/workflow.ts

```diff
@@ -0,0 +1,370 @@
+          all: [
+            { has: { field: 'object', regex: `^${localNamespace}$` } },
+            { has: { field: 'property', regex: `^${oldFunctionName}$` } }
+          ]
+        }
+      }
+    });
+
+    if (!usages.length) return [];
+
+    const edits = usages
+      .map(u => u.field('function')?.replace(`${newNamespace}.${newImportFunction}`))
+      .filter(Boolean) as Edit[];
+
+    edits.push(declaration.replace(`const ${newNamespace} = await import('${newImportModule}');`));
+    return edits;
+  }
+
+  // Case 2: `const { ... } = await import(...)`
+  if (idNode?.kind() === 'object_pattern') {
+    const specifiers = idNode.children().filter(
+      c => c.kind() === 'pair_pattern' || c.kind() === 'shorthand_property_identifier_pattern'
+    );
+
+    let targetSpecifier: SgNode | null = null;
+    let localFunctionName: string | null | undefined = null;
+    let isAliased = false;
+
+    for (const spec of specifiers) {
+      const keyNode = spec.kind() === 'pair_pattern' ? spec.field('key') : spec;
+      if (keyNode?.text() === oldFunctionName) {
+        targetSpecifier = spec;
+        isAliased = spec.kind() === 'pair_pattern';
+        localFunctionName = isAliased ? spec.field('value')?.text() : keyNode.text();
+        break;
+      }
+    }
+
+    if (!localFunctionName || !targetSpecifier) return [];
+
+    const edits: Edit[] = [];
+    if (!isAliased) edits.push(...findAndReplaceUsages(rootNode, localFunctionName, newImportFunction));
+
+    const newImportSpecifier = isAliased
+      ? `{ ${newImportFunction}: ${localFunctionName} }`
+      : `{ ${newImportFunction} }`;
+
+    const newImportStmt = `const ${newImportSpecifier} = await import('${newImportModule}');`;
+
+    const otherSpecifiers = specifiers.filter(s => s !== targetSpecifier);
+    if (otherSpecifiers.length) {
+      const remaining = `const { ${otherSpecifiers.map(s => s.text()).join(', ')} } = await import('${oldImportModule}');`;
+      edits.push(declaration.replace(`${remaining}${EOL}${newImportStmt}`));
+    } else {
+      edits.push(declaration.replace(newImportStmt));
+    }
+
+    return edits;
+  }
+
+  return [];
+}
+
+export default function transform(root: SgRoot): string | null {
+  const rootNode = root.root();
+  const allEdits: Edit[] = [];
+  let wasTransformed = false;
+
+  const sources: [SgNode[] | undefined, (n: SgNode, r: SgRoot) => Edit[]][] = [
+    // @ts-ignore
+    [getNodeRequireCalls(root, 'crypto'), handleRequire],
+    // @ts-ignore
+    [getNodeImportStatements(root, 'crypto'), handleStaticImport],
+    // @ts-ignore
+    [getNodeImportCalls(root, 'crypto'), handleDynamicImport],
+  ];
+
+  for (const [nodes, handler] of sources) {
+    for (const node of nodes || []) {
+      const edits = handler(node, root);
+      if (edits.length) {
+        wasTransformed = true;
+        allEdits.push(...edits);
+      }
+    }
+  }
+
+  return wasTransformed ? rootNode.commitEdits(allEdits) : null;
```

Comment:
wasTranformed var and be replaced by `!allEdits.lenght`.

Also use this structure instead of ternary:

```js
if (!allEdits.lenght) return null

return rootNode.commitEdits(allEdits)
```